tournamentcontrol/competition/tests/test_logo_fields.py

```diff
@@ -0,0 +1,240 @@
+        
+        # Create a simple PNG image
+        img = Image.new("RGB", (100, 100), color="red")
+        img_bytes = io.BytesIO()
+        img.save(img_bytes, format="PNG")
+        img_bytes.seek(0)
+        cls.png_data = img_bytes.read()
+        
+        # Create a simple JPEG image
+        img = Image.new("RGB", (100, 100), color="blue")
+        img_bytes = io.BytesIO()
+        img.save(img_bytes, format="JPEG")
+        img_bytes.seek(0)
+        cls.jpg_data = img_bytes.read()
+        
+        # Create a simple SVG (text-based)
+        cls.svg_data = b'<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100"></svg>'
+        
+        # Create invalid file (PDF)
+        cls.pdf_data = b"%PDF-1.4 fake pdf content"
+
+    def test_logo_fields_optional(self):
+        """Verify logo fields are optional (blank=True, null=True)."""
+        # ImageField.name is None when no file is uploaded
+        self.assertIsNone(self.competition.logo_colour.name)
+        self.assertIsNone(self.competition.logo_monochrome.name)
+
+    def test_competition_logo_via_admin(self):
+        """Test that Competition edit view includes logo fields."""
+        namespace = self.competition._get_admin_namespace()
+        args = self.competition._get_url_args()
+        
+        with self.login(self.superuser):
+            response = self.get(f"{namespace}:edit", *args)
+            self.response_200()
+            self.assertContains(response, "logo_colour")
+            self.assertContains(response, "logo_monochrome")
+
+    def test_club_logo_via_admin(self):
+        """Test that Club edit view includes logo fields."""
+        namespace = self.club._get_admin_namespace()
+        args = self.club._get_url_args()
+        
+        with self.login(self.superuser):
+            response = self.get(f"{namespace}:edit", *args)
+            self.response_200()
+            self.assertContains(response, "logo_colour")
+            self.assertContains(response, "logo_monochrome")
+
+    def test_season_logo_via_admin(self):
+        """Test that Season edit view includes logo fields."""
+        namespace = self.season._get_admin_namespace()
+        args = self.season._get_url_args()
+        
+        with self.login(self.superuser):
+            response = self.get(f"{namespace}:edit", *args)
+            self.response_200()
+            self.assertContains(response, "logo_colour")
+            self.assertContains(response, "logo_monochrome")
+
+    def test_division_logo_via_admin(self):
+        """Test that Division edit view includes logo fields."""
+        namespace = self.division._get_admin_namespace()
+        args = self.division._get_url_args()
+        
+        with self.login(self.superuser):
+            response = self.get(f"{namespace}:edit", *args)
+            self.response_200()
+            self.assertContains(response, "logo_colour")
+            self.assertContains(response, "logo_monochrome")
```

Comment:
These should be calling POST and sending the image file data, and assert it was redirected.

Use `InMemoryStorage` via `override_settings` and check the file (including contents) exists at the expected path.